lib/project-data.ts

```diff
@@ -90,6 +90,14 @@ export class ProjectData implements IProjectData {
 		this.errorInvalidProject(projectDir);
 	}
 
+	public initializeProjectDataSafe(projectDir?: string): void {
```

Comment:
After a discussion we decided it is better to try/catch when needed, than adding a method.